install.sh

```diff
@@ -161,28 +141,131 @@ __atuin_install_unsupported(){
 		esac
 	done
 }
+__atuin_check_gcc(){
+    OS=$(__get_os)
+    if command -v gcc &> /dev/null; then
+        echo "gcc is installed"
+    else
+        echo "gcc not installed. Please install gcc."
+        exit
+    fi
+}
+
+__get_os() {
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    OS="darwin"
+  elif [ -f "/etc/os-release" ]; then
+    OS="$(grep -Po '(?<=^ID=).*$' /etc/os-release | tr '[:upper:]' '[:lower:]')" 2>/dev/null
+  elif [ -f "/etc/debian_version" ]; then
+    OS="debian"
+  elif [ -f "/etc/centos-release" ]; then
+    OS="centos"
+  elif [ -f "/etc/manjaro-release" ]; then
+    OS="manjarolinux"
+  elif [ -f "/etc/fedora-release" ]; then
```

Comment:
I don’t know how relevant it could be and how wide you want to support Linux distros, but there is also `/etc/redhat-release`; however, be aware that it’s also present on Fedora systems. The same should be true with `ubuntu-release` and `debian-release` but i still have to check that. There might also be `mint-release` and who knows what… :roll_eyes:

Reply:
I think the best way to resolve this would be to gather the actual dependencies needed and check for those but I don't know if that's worth it unless we start involving binaries. 

I can certainly edit it to incorporate other distros as needed though.

 Thoughts?